### PR TITLE
Update Metric aggregation Callback

### PR DESCRIPTION
**Pull request by Dokholyan**
## Before submitting (checklist)

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [ ] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [ ] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [ ] Did you check the docs with `make check-docs`?
- [ ] Did you write any new necessary tests?
- [ ] Did you check that your code passes the unit tests `pytest .` ? 
- [ ] Did you add your new functionality to the docs?
- [ ] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

PS
- [x] I know, that I could [join slack](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for pull request discussion.

### CODE REVIEW

File: catalyst/callbacks/metric.py [hunk 1]
```diff
@@ -309,9 +309,11 @@ def __init__(
         prefix: str,
         metrics: Union[str, List[str], Dict[str, float]] = None,
         mode: str = "mean",
+        aggregation_function: Optional[Callable] = None,
```
Comment (Scitator) [inline catalyst/callbacks/metric.py]: could we rename it to `aggregation_fn`? a question of codestyle

File: catalyst/callbacks/metric.py [hunk 2]
```diff
@@ -309,9 +309,11 @@ def __init__(
...
         scope: str = "batch",
         multiplier: float = 1.0,
     ) -> None:
+        # ToDo Example with aggregation_function
```
Comment (Scitator) [inline catalyst/callbacks/metric.py]: yup, looking forward to them

File: catalyst/callbacks/metric.py [hunk 3]
```diff
@@ -342,6 +351,13 @@ def __init__(
                     "the metrics must be specified "
                     "and must be a dict"
                 )
+        elif mode == "custom":
```
Comment (Scitator) [inline catalyst/callbacks/metric.py]: 🤔  could we use mode as a `Callable` too? I mean, why do we need `aggregation_function` and `mode` if we could leave only `mode` and use it as `aggregation_function`?

File: catalyst/callbacks/metric.py [hunk 4]
```diff
@@ -386,9 +404,12 @@ def _preprocess(self, metrics: Any) -> List[float]:
             result = list(metrics.values())
         return result
 
-    def _process_metrics(self, metrics: Dict):
-        metrics_processed = self._preprocess(metrics)
-        metric_aggregated = self.aggregation_fn(metrics_processed)
+    def _process_metrics(self, metrics: Dict, runner: "IRunner"):
```
Comment (Scitator) [inline catalyst/callbacks/metric.py:415]: why do we need `runner` here?

**Review by Scitator — DISMISSED** [review]

File: catalyst/callbacks/metric.py (hunk 1 above)
Comment (Dokholyan) [reply catalyst/callbacks/metric.py]: Ok

File: catalyst/callbacks/metric.py (hunk 4 above)
Comment (Dokholyan) [reply catalyst/callbacks/metric.py:415]: Because it is used in  "self.aggregation_fn(metrics, runner)" if mode == 'custom'

Comment (Dokholyan) [reply catalyst/callbacks/metric.py:415]: Because it is used in "self.aggregation_fn(metrics, runner)" if mode == 'custom'

Comment (Scitator) [reply catalyst/callbacks/metric.py:415]: and why do we need `runner` for `aggregation_fn`?

Comment (Dokholyan) [reply catalyst/callbacks/metric.py:415]: For custom aggregation, for example, to use epoch

Comment (Scitator) [reply catalyst/callbacks/metric.py:415]: 🤔  in this case, we can just pass runner.... need to think about it – how we could do better

**Pull request closed without merging**